packages/SwingSet/src/liveslots/collectionManager.js

```diff
@@ -62,6 +63,18 @@ function pattEq(p1, p2) {
   return compareRank(p1, p2) === 0;
 }
 
+function throwNotDurable(value, slotIndex, serializedValue) {
+  assert.fail(
+    X`value is not durable: ${value} at slot ${slotIndex} of ${decodeToJustin(
+      {
+        body: JSON.parse(serializedValue.body),
+        slots: serializedValue.slots,
+      },
+      true,
+    )}`,
+  );
```

Comment:
It's a lot more ceremony, but I think it is worth it. @FUDCo , please weigh in. Thanks.

```suggestion
  const body = JSON.parse(serializedValue.body);
  let encodedValue;
  try {
    encodedValue = decodeToJustin(
      {
        body,
        slots: serializedValue.slots,
      },
      true,
    );
  } catch (justinError) {
    const err = assert.error(
      X`value is not durable: ${value} at slot ${slotIndex} of ${serializedValue}`,
    );
    assert.note(
      err,
      X`decodeToJustin(${serializedValue}) threw ${justinError} `,
    );
    throw err;
  }
  assert.fail(
    X`value is not durable: ${value} at slot ${slotIndex} of ${encodedValue}`,
  );
```

Reply:
See https://github.com/endojs/endo/issues/1185

I'll fix that. But since that's in endo, we can't assume the fix here until the next cycle of the release dance completes. An error in reporting an error is rarely more important to report than the error we were originally trying to report in the first place. Diagnostics reporting errors should try to be robust in this sense. The above change reports both errors, but emphasizes the error we were trying to report.

Reply:
I just pushed the change I suggest above, so resolving. Of course, unresolve and continue the conversation if there's any issue.